src/main/scala/vsys/blockchain/state/opcdiffs/BasicOpcDiff.scala

```diff
@@ -128,6 +128,14 @@ object BasicOpcDiff extends OpcDiffer {
     val Not         = BasicTypeVal(14, 3, Seq(1),    (b, d) => not(d(b(1))))
   }
 
+  private def bigintBytes (bytes: Array[Byte]): Array[Byte] = {
+    val x = bytes.dropRight(1).dropWhile(i => i == 0) ++ bytes.takeRight(1)
```

Comment:
This function seems to have the same effect as BigInt(bytes).toByteArray

Reply:
it seems that BigInt apply java BigInteger to transfer the format
```scala
  /** Translates a byte array containing the two's-complement binary
   *  representation of a BigInt into a BigInt.
   */
  def apply(x: Array[Byte]): BigInt =
    new BigInt(new BigInteger(x))
```

Based on the BigInteger source code(http://hg.openjdk.java.net/jdk/jdk11/file/tip/src/java.base/share/classes/java/math/BigInteger.java), it seems that we can use
```scala
if(BigInt(bytes).isValidInt) Ints.toByteArray(BigInt(bytes).toInt)
if(BigInt(bytes).isValidLong) Longs.toByteArray(BigInt(bytes).toInt)

```
here?
@zl730 will it causes extra calculation here?

Reply:
Pure byte array operations here avoiding BigInt object creation for lower time and space complexity